### ISSUE TITLE
Can it accept array as JSON event

### ISSUE THREAD

**Issue by iamdionysus** [apex/apex#759]
Considering when I pass `["hello world"]`, I get `Error: parsing response: json: cannot unmarshal array into Go value of type map[string]interface {}`, it looks like apex can not accept array as JSON event at the moment. However, based on official [json](http://json.org) structure, it says

> JSON is built on two structures:

> A collection of name/value pairs. In various languages, this is realized as an object, record, struct, dictionary, hash table, keyed list, or associative array.
> An ordered list of values. In most languages, this is realized as an array, vector, list, or sequence. 

In that case, shouldn't `["hello world"]` be acceptable JSON event?

**Comment by tj** [comment]
Yep the signature is `input json.RawMessage, *Context` so you can unmarshal the `input` to an array if you wanted :D

**Comment by iamdionysus** [comment]
Thanks for the reply. I think I have to be more clear. When I try 

    $ echo -n '["hello world"]' | apex invoke hello

I get `Error: parsing response: json: cannot unmarshal array into Go value of type map[string]interface {}` because the `apex/cmd/apex/invoke/invoke.go` expects the input should be `map[string]` like below. My question is can I pass JSON array as valid argument in apex cli.

```go
// Run command.
func run(c *cobra.Command, args []string) error {
	stats.Track("Invoke", map[string]interface{}{
		"has_alias": alias != "",
		"has_logs":  includeLogs,
	})
```

**Comment by tj** [comment]
ahh from the cli, right my bad, maybe not at the moment

**Comment by iamdionysus** [comment]
Thanks for the review. Sorry for the lack of details.

**Comment by polds** [comment]
We've actually run into this before and had to put API Gateway in front of Lambda to translate the array into an object. Might not be in your use case but thought I'd share.

This is the simple mapping template we used:

```
{"key" : $input.json('$')}
```

**Comment by tj** [comment]
TBH I'd probably pass objects always anyway, just because it's more future-proof API-wise, and more self-descriptive, like what does that array represent, but it's still a bug :D
